sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/RxDocumentClientImpl.java

```diff
@@ -4123,6 +4122,16 @@ public synchronized void enableThroughputControlGroup(ThroughputControlGroupInte
         this.throughputControlStore.enableThroughputControlGroup(group);
     }
 
+    @Override
+    public Flux<OpenConnectionResponse> openConnectionsAndInitCaches(String containerLink) {
+        checkArgument(StringUtils.isNotEmpty(containerLink), "Argument 'containerLink' should not be null nor empty");
+
+        DocumentClientRetryPolicy retryPolicyInstance = this.resetSessionTokenRetryPolicy.getRequestPolicy();
```

Comment:
Wrapped within client retry policy, so can have retries when failed to get the addresses back from gateway. Does it make sense?

Reply:
Yes, we need the gateway addresses anyway at some point, better to retry here and fail if we can't retrieve them.

Reply:
I do not think we should reuse the clientRetryPolicy, because it may cause the SDK to failover to a different region. 
I want to limit the behavior of the openAsync's behavior today, so added a new OpenConnectionAndInitCachesRetryPolicy to only retry on 1. gateway timeout for address resolution 2. 429